platform_logs/test/test.js

```diff
@@ -1,11 +1,357 @@
-const assert = require('chai').assert;
+const chai = require('chai');
+const assert = chai.assert;
+const expect = chai.expect;
 const index = require('../index');
+const awsContext = require('aws-lambda-mock-context');
+const sinon = require('sinon');
+const logger = require("../components/logger.js");
+const request = require('request');
 
-describe('Sample', function() {
-  it('tests handler', function(done) {
+describe('platform_logs', function() {
+  var event, context, callback, spy, stub, logStub, logMessage, errorMessage, errorType;
 
-    // Add your test cases here.
-    assert(true);
-    done();
+  //setup a helper function to check for expected outputs given different input parameters
+  var inputValidation = (eventProp, eventProp2, newValue, errMessage, errType) => {
+    //if there is a second prop defined, than the value to be changed is in a nested object
+    if(eventProp2){
+      event[eventProp][eventProp2] = newValue;
+    }
+    else if(eventProp){
+      event[eventProp] = newValue;
+    }
+    else{
+      event = newValue;
+    }
+    //check if handler returns error notification with expected error type and message
+    var handlerResponse = index.handler(event, context, callback);
+    var bool = handlerResponse.includes(errMessage) && handlerResponse.includes(errType);
+    return bool;
+  }
+
+  //setup additional helper function for validating multiple values
+  var multipleValidation = (prop1, prop2, valueArray, errMessage, errType) => {
+    var allChecks = true;
+    //allChecks stays true if handler() returns expected error for every value assigned
+    for(i in valueArray){
+      var bool = inputValidation(prop1, prop2, valueArray[i], errMessage, errType);
+      if(!bool){
+        allChecks = false;
+        break;
+      }
+    }
+    return allChecks;
+  }
+
+  beforeEach(function(){
+    //setup spy to wrap async/extraneous functions
+    spy = sinon.spy();
+    event = {
+      "stage" : "test",
+      "body_get":{},
+      "body_put":{
+        "endpoints" : {
+          "dev": "newval"
+        }
+      },
+      "body":{
+        "environment": "test",
+        "service": "dAnCe",
+        "domain": "bA11r00m",
+        "category": "api",
```

Comment:
category is also a variable to this API. One of the other values is function, so please add test to validate that.

Reply:
Also looks like we have a bug where our code moves forward when it is none of those two values, so please add a unit test for that too.